File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/functions/FirstFunctionFragment.java
package com.laushkina.anastasia.psychosophy.view.functions;

import android.graphics.drawable.Drawable;
import android.text.Html;
import android.text.Spanned;

import com.laushkina.anastasia.psychosophy.R;
import com.laushkina.anastasia.psychosophy.domain.test.PsychoFunction;

public class FirstFunctionFragment extends FunctionsDescriptionFragment {

    @Override
    public Drawable getImage() {
        return getResources().getDrawable(R.drawable.ic_hammer);
    }

    @Override
    public Drawable getImage(PsychoFunction function) {
        switch (function) {
            case Emotion: return getResources().getDrawable(R.drawable.ic_hammer_emotion);
            case Logic: return getResources().getDrawable(R.drawable.ic_hammer_logic);
            case Physics: return getResources().getDrawable(R.drawable.ic_hammer_physics);
            case Will: return getResources().getDrawable(R.drawable.ic_hammer_will);
            default: throw new AssertionError();
        }
    }

    @Override
    public CharSequence getFunctionDescription() {
        return Html.fromHtml(getString(R.string.first_function_description));
    }


    @Override
    public Spanned getFullDescription(PsychoFunction function) {
        switch (function) {
            case Emotion: return Html.fromHtml(getResources().getString(R.string.first_emotion_description));
            case Logic: return Html.fromHtml(getResources().getString(R.string.first_logic_description));
            case Physics: return Html.fromHtml(getResources().getString(R.string.first_physics_description));
            case Will: return Html.fromHtml(getResources().getString(R.string.first_will_description));
            default: throw new AssertionError();
        }
    }

    @Override
    public String getEmotionTitle() {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Emotion, getResources());
    }
    @Override
    public String getEmotionShortTitle() {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Emotion, getResources());
    }

    @Override
    public String getLogicShortTitle() {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Logic, getResources());
    }

    @Override
    public String getLogicTitle() {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Logic, getResources());
    }

    @Override
    public String getPhysicsShortTitle() {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Physics, getResources());
    }

    @Override
    public String getPhysicsTitle() {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Physics, getResources());
    }

    @Override
    public String getWillShortTitle() {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Will, getResources());
    }

    @Override
    public String getWillTitle() {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Will, getResources());
    }

}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/functions/FirstFunctionFragment.kt
package com.laushkina.anastasia.psychosophy.view.functions

import android.graphics.drawable.Drawable
import android.text.Html
import com.laushkina.anastasia.psychosophy.R
import com.laushkina.anastasia.psychosophy.domain.test.PsychoFunction

class FirstFunctionFragment: FunctionsDescriptionFragment() {
    override fun getImage(): Drawable {
        return resources.getDrawable(R.drawable.ic_hammer)
    }

    override fun getImage(function: PsychoFunction): Drawable {
        return when (function) {
            PsychoFunction.Emotion -> resources.getDrawable(R.drawable.ic_hammer_emotion)
            PsychoFunction.Logic -> resources.getDrawable(R.drawable.ic_hammer_logic)
            PsychoFunction.Physics -> resources.getDrawable(R.drawable.ic_hammer_physics)
            PsychoFunction.Will -> resources.getDrawable(R.drawable.ic_hammer_will)
        }
    }

    override fun getFunctionDescription(): CharSequence {
        return Html.fromHtml(getString(R.string.first_function_description))
    }

    override fun getFullDescription(function: PsychoFunction): CharSequence {
        return when (function) {
            PsychoFunction.Emotion -> Html.fromHtml(resources.getString(R.string.first_emotion_description))
            PsychoFunction.Logic -> Html.fromHtml(resources.getString(R.string.first_logic_description))
            PsychoFunction.Physics -> Html.fromHtml(resources.getString(R.string.first_physics_description))
            PsychoFunction.Will -> Html.fromHtml(resources.getString(R.string.first_will_description))
        }
    }

    override fun getEmotionShortTitle(): String {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Emotion, resources)
    }

    override fun getEmotionTitle(): String {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Emotion, resources)
    }

    override fun getLogicShortTitle(): String {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Logic, resources)
    }

    override fun getLogicTitle(): String {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Logic, resources)
    }

    override fun getPhysicsShortTitle(): String {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Physics, resources)
    }

    override fun getPhysicsTitle(): String {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Physics, resources)
    }

    override fun getWillShortTitle(): String {
        return FunctionTitleGetter.getFirstFunctionShortTitle(PsychoFunction.Will, resources)
    }

    override fun getWillTitle(): String {
        return FunctionTitleGetter.getFirstFunctionTitle(PsychoFunction.Will, resources)
    }

}